x-pack/plugins/apm/public/components/alerting/ui_components/chart_preview/chart_preview_helper.tsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiLoadingChart } from '@elastic/eui';
+import { EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { niceTimeFormatter } from '@elastic/charts';
+import { min as getMin, max as getMax } from 'lodash';
+import { Coordinate } from '../../../../../typings/timeseries';
+
+export const TIME_LABELS = {
+  s: i18n.translate('xpack.apm.alerts.timeLabels.seconds', {
+    defaultMessage: 'seconds',
+  }),
+  m: i18n.translate('xpack.apm.alerts.timeLabels.minutes', {
+    defaultMessage: 'minutes',
+  }),
+  h: i18n.translate('xpack.apm.alerts.timeLabels.hours', {
+    defaultMessage: 'hours',
+  }),
+  d: i18n.translate('xpack.apm.alerts.timeLabels.days', {
+    defaultMessage: 'days',
+  }),
+};
```

Comment:
Like mentioned elsewhere: I'm surprised this is implemented in the APM app. I'd expect this to be solved generically elsewhere

Reply:
I have seen similar implementation in other plugins with some variations. Maybe it could be moved to common place where all plugins can access it. I will look into it.